app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,65 @@
+class ExpensesController < ApplicationController
+
+  require 'csv'
+  
+  def index
+    @tour = Tour.find(params[:id])
+    @expenses = Expense.where(tour_id:params[:id])
+
+  end
+
+  def new
+    @tour = Tour.find(params[:id])
+    @expense = Expense.new
+  end
+
+  def create
+    
+    post_data = params.require(:expense).permit(
+      :date,
+      :city,
+      :cost,
+      :state,
+      :category,
+      :tour_id,
+    )
+
+    ap "post data paraaams"
+    ap post_data
+
+    @expense = Expense.new(post_data)
+    
+
+    if @expense.save
+
+      @url = '/tours/1/expenses'
+
+      redirect_to @url
```

Comment:
Did you want to update this to be dynamic as we did in the `import` method?

Reply:
will leave to save time